shell/platform/linux/fl_standard_message_codec_test.cc

```diff
@@ -1077,6 +1090,178 @@ TEST(FlStandardMessageCodecTest, DecodeUnknownType) {
+  uint8_t type = 129;
+  g_byte_array_append(buffer, &type, sizeof(uint8_t));
+  return TRUE;
+}
+
+static gboolean fl_test_standard_message_codec_write_value(
+    FlStandardMessageCodec* codec,
+    GByteArray* buffer,
+    FlValue* value,
+    GError** error) {
+  if (fl_value_get_type(value) == FL_VALUE_TYPE_CUSTOM &&
+      fl_value_get_custom_type(value) == 128) {
+    return write_custom_value1(codec, buffer, value, error);
+  } else if (fl_value_get_type(value) == FL_VALUE_TYPE_CUSTOM &&
+             fl_value_get_custom_type(value) == 129) {
+    return write_custom_value2(codec, buffer, value, error);
+  } else {
+    return FL_STANDARD_MESSAGE_CODEC_CLASS(
+               fl_test_standard_message_codec_parent_class)
+        ->write_value(codec, buffer, value, error);
+  }
+}
+
+static FlValue* read_custom_value1(FlStandardMessageCodec* codec,
+                                   GBytes* buffer,
+                                   size_t* offset,
+                                   GError** error) {
+  uint32_t length;
+  if (!fl_standard_message_codec_read_size(codec, buffer, offset, &length,
+                                           error)) {
+    return nullptr;
+  }
+  if (*offset + length > g_bytes_get_size(buffer)) {
+    g_set_error(error, FL_MESSAGE_CODEC_ERROR,
+                FL_MESSAGE_CODEC_ERROR_OUT_OF_DATA, "Unexpected end of data");
+    return nullptr;
+  }
+  FlValue* value = fl_value_new_custom(
+      128,
+      g_strndup(static_cast<const gchar*>(g_bytes_get_data(buffer, nullptr)) +
+                    *offset,
+                length),
+      g_free);
+  *offset += length;
+
+  return value;
+}
+
+static FlValue* read_custom_value2(FlStandardMessageCodec* codec,
+                                   GBytes* buffer,
+                                   size_t* offset,
+                                   GError** error) {
+  return fl_value_new_custom(129, nullptr, nullptr);
+}
+
+static FlValue* fl_test_standard_message_codec_read_value_of_type(
+    FlStandardMessageCodec* codec,
+    GBytes* buffer,
+    size_t* offset,
+    int type,
+    GError** error) {
+  if (type == 128) {
+    return read_custom_value1(codec, buffer, offset, error);
+  } else if (type == 129) {
+    return read_custom_value2(codec, buffer, offset, error);
+  } else {
+    return FL_STANDARD_MESSAGE_CODEC_CLASS(
+               fl_test_standard_message_codec_parent_class)
+        ->read_value_of_type(codec, buffer, offset, type, error);
+  }
+}
+
+static void fl_test_standard_message_codec_class_init(
+    FlTestStandardMessageCodecClass* klass) {
+  FL_STANDARD_MESSAGE_CODEC_CLASS(klass)->write_value =
+      fl_test_standard_message_codec_write_value;
+  FL_STANDARD_MESSAGE_CODEC_CLASS(klass)->read_value_of_type =
+      fl_test_standard_message_codec_read_value_of_type;
+}
+
+static void fl_test_standard_message_codec_init(
+    FlTestStandardMessageCodec* self) {
+  // The following line suppresses a warning for unused function
```

Comment:
I'm assuming _init functions remain required even if they do nothing? 

I'd have thought this was macro-based and that those would have been expanded during analysis -- is that not the case?

Reply:
A macro generates the call to the init functions, so they have to exist. Every GObject must have both init and class_init, so I think this is so standard there's no need to explicitly comment on them.

`FL_IS_TEST_STANDARD_MESSAGE_CODEC` calls a function generated from another macro and is normally used at the top of each method the object provides. Since this class has no (non overridden) methods it's never used and the compiler flags are set to generate an error for unused functions.